### DISCUSSION TITLE
With Jamulus, plan to add or move-to Linux Ubuntu ?

### DISCUSSION THREAD

**Discussion by pcar75** [jamulussoftware/jamulus#2518] [General]
Bonjour,

- Ubuntu seems a nice stable Linux redistributor
- My JamulusOS redistribution uses XUbuntu (Ubuntu with Xfce desktop) 20.04.1 LTS from a USB key.  Originally with Jamulus 3.6.2, now with 3.8.2.
- With my Windows10 tower, I use Jamulus to listen to others, but reboot to the Ubuntu Jamulus to play with others (lower overall delay).
- I am considering installing Ubuntu on my SSD C: drive (repartitionned). as a dual boot system, and given uncertain Windows11 and purchase model, possibly migrating completely to Linux later.
- Current Ubuntu desktop has many flavors each with many versions (varying stability and support).
- XUbuntu 22.04 could be interesting when it becomes LTS (2022 ?) ref.: https://linuxconfig.org/ubuntu-linux-download
- Ubuntu Studio could also be interesting. However, it is in the process of moving from the Xfce desktop to the KDE Plasma desktop and I'd rather get something stable and (mostly) future-proof and deepen my knowledge of one environment sooner than later. 
ref.: https://ubuntustudio.org/
- Since the Studio version is migrating to KDE Plasma, KUbuntu 22.04 could also be interesting when it becomes LTS (2022 ?) 
- The Studio version would possibly be easier to configure for audio, but it would probably require me to install usual standard softwares (ex.LibreOffice) or other sw (ex. Reaper) , but these may not be adapted to the KDE Plasma desktop environment.
- Then there are discussions of moving Jamulus from Qt5.xxx to Qt6.xxx and I presume that would be a lot of other desktop environment interdependencies.
- As an ex-developper I've lived through the Windows .dll wars and am in no hurry to start that again with linux components and version dependencies.

Thanks in advance for any constructive lines of thought ?

**Comment by ann0see** [comment]
I'm not sure if I understand you correctly, but I'm working with both OS and in general prefer Jamulus on Linux due to better JACK support. 
Linux is probably quicker in all terms and didn't give me a kernel panic (=BSOD) yet.
Yes, I'd recommend to use Linux, but it's totally a personal choice.

**Reply by pcar75** [reply]
Thanks, my hesitations are due to which flavor and desktop environment would best suit me given I want to do audio work but also desktop/business type things and may migrate entirely to Linux in the future if I can use it with minimal fuss (libraries chasing, recompile, etc) , 
The variety of linux distributions/flavors/version, along with human interface layers (Gnome, KDE, Xfce, X, etc.) each with different requirements for applications and their components is bewildering coming from the Windows world (I confess some ex-developper experience on MicroVax Ultrix ... in the preceding millennium , no GUI   :-/

**Reply by ann0see** [reply]
I‘m afraid I can’t help you with choosing a flavour. I went with Debian + GNOME but Ubuntu might have more end user documentation. I didn’t have to do much frickling with standard software & hardware.

**Comment by dcorson-ticino-com** [comment]
To tell you the truth, Windows isn't perfect, but it is stable and I see no difference in Jamulus' performance between Ubuntu and Win10.  Try and find what is slowing down your Win10 installation (my suspicion is always the ASIO driver, which, of course, MS has no control over). 
I stopped using Linux when I started noticing the tremendous amount of time I was spending just keeping things working.  
I see computers as tools to get things done.  If I spend more time keeping the tool running it is no good for me.
There are certainly domains where Linux is the way to go, but to run a browser, LibreOffice and a Jamulus client Win10 just works (I have never had a crash with Jamulus under Win10 either).

**Reply by pcar75** [reply]
Well, Windows10 is stable now, but given MS record, I would definitly wait a few updates before giving 11 a chance (if ever) and payment.  Windows11 has been enhanced for foreground task (i.e. visual), so may not be the best for audio and background tasks (Jack, JackRoute, REARoute, etc.),  Also MS seems to have unilaterally tweaked the IP stack ; Ref. "3.8.0 on Windows 10 : problem with loopback (127.0.0.1)" #1855

My USB key Ubuntu Jamulus is slow to load (that's why I'm considering installing it alongside Win10 on My SSD drive) , but tests after tests, it shaves off anywhere from 7 to 20 ms off the overall delay (depending on the public server) better than Win10. Given the distances in the americas, it's important.
I never had a Blue Screen Of Death with Ubuntu either  ;-)

**Reply by dcorson-ticino-com** [reply]
Yeh, Win11 looks like a version to skip.  There seem to be lots of performance reducing "enhancements".
I am not changing for the simple reason that it is no longer possible to put the task bar anywhere other than at the bottom. On screens that are already way to wide (made to watch films, but not to work on) putting the taskbar on the left or right was at least a useful thing to do to use that area.
But we know MS in the mean time.  They sometimes try to jump too far and then correct the mistakes in the next version.
On Linux, 'though the confusion seems to increase every year.
As I said, talk to your interface manufacturer about getting a better ASIO driver.  I see no (0) difference between Win10 and a USB Ubuntu Jamulus with a NI Komplete 2 interface..

**Reply by rdica** [reply]
@pcar75 You already know my thoughts and position on Linux/Jamulus use. I'm not sure you're aware thiough that if using Ubuntu or UbuntuStudio, you can always install other desktop environments besides what the default is. Nothing says you must use KDE if thats the default for the 22.04 LTS version of UbuntuStudio, for example. You can install XFCE desktop afterwards. The software pkgs for Ubuntu are the same as UbuntuStudio.

**Reply by pcar75** [reply]
@rdica Can I use both **simultaneously** (ex. KDE for Jamulus and most audio in UStudio  and Xfce for xxxOffice) ?  On a single monitor / display ?

**Reply by rdica** [reply]
LibreOffice runs on any desktop environment. Of course you can't run two desktop envs simultaneously as the same user, but the choice of desktop does not restrict what programs you can run in it. Jamulus works as well under XFCE as it does under KDE, as does LibreOffice.
Choose a desktop you like, and go with it :)
Does that answer your question?

**Reply by pcar75** [reply]
My question, Yes ; my hopes, No !
Thanks !

**Reply by pljones** [reply]
If you've got multiple monitors, you might be able to run multiple X instances and log into each as a separate user with a separate desktop environment.  But basically, the desktop is what you see on your monitor in an X instance for a user, so if you want multiple desktop environments, you need multiple of the other parts, too.

**Comment by pcar75** [comment]
Thanks all for your comments !
